store/src/java/com/zimbra/cs/ldap/unboundid/UBIDLdapFilterFactory.java

```diff
@@ -860,6 +863,60 @@ public ZLdapFilter accountsWithDocumentEditing() {
                         FILTER_ACCOUNTS_WITH_DOCUMENT_EDITING));
     }
 
+    @Override
+    public ZLdapFilter accountsByCosesAndFeatureCheck(List<String> cosIds, String ldapAttribute) {
+        List<Filter> filters = Lists.newArrayList();
+        for (String cosId : cosIds) {
+            filters.add(Filter.createEqualityFilter(Provisioning.A_zimbraCOSId, cosId));
+        }
+        filters.add(Filter.createNOTFilter(Filter.createPresenceFilter(Provisioning.A_zimbraCOSId)));
+        return new UBIDLdapFilter(
+                FilterId.ACCOUNTS_BY_COSES_AND_LDAP_FEATURE_CHECK,
+                Filter.createANDFilter(
+                        FILTER_ALL_NON_SYSTEM_ACCOUNTS,
+                        Filter.createNOTFilter(
+                            Filter.createORFilter(
+                                Filter.createEqualityFilter(ldapAttribute, LdapConstants.LDAP_TRUE),
+                                Filter.createEqualityFilter(ldapAttribute, LdapConstants.LDAP_FALSE))),
+                        Filter.createORFilter(filters)));
+    }
```

Comment:
Use  `stream` for creating filters in `accountsByCosesAndFeatureCheck` to reduces boilerplate code and improved readability.

Reply:
Done